### ISSUE TITLE
[feature] make URI class dual-life (distribution)

### ISSUE THREAD

**Issue by rwp0** [Perl/perl5#18378]
can [URI](https://metacpan.org/release/URI) distribution be made dual-life also existing as as core module(s)?

i think it is being fairly used frequently enough for a consideration in parsing various protocol schema specifications (such as SSH and SFTP) in a more convenient object-oriented way

i believe it has no external dependencies and can complement `File::` modules in core

**Comment by Grinnz** [comment]
The terminology for this is to make the module dual-life (living in core and CPAN).

**Comment by Leont** [comment]
I do not believe URI is of sufficiently broad utility to warrant inclusion in core, even if it's of deep utility in certain not-uncommon niches.

**Comment by rwp0** [comment]
> I do not believe URI is of sufficiently broad utility to warrant inclusion in core, even if it's of deep utility in certain not-uncommon niches.

list of additional reasons pro having URI in the standard distribution (as noticed from my personal perspective as a programming language user):

- Perl is good in parsing, why not parse URI's in standard way (having a library implementation)
- i also use a separate `URI::db` distribution for `DBI` DSN's (it having database, and system administration uses)
- many other high level languages have such utility libraries built-in (for good reasons, i believe)
- it has multitude of CPAN distribution reverse dependencies (people use it valuing the convenience provided thereby)
- HTTP, and other protocol schemata are essential in real-life (eg. in small-scale scripting encountered often)
- complete systems are preferred by many over distributed ones (though moderation to be watched, i assume)

i think it might have more use than many other modules in core, and can replace, or extend other similar ones given the lack thereof (URI handling missing in a programming language system)

---

DBI, DateTime, URI (namespaces not needed) etc is Perl, as every real (experienced, tasked, attempted) user of the language (or the interpreter) knows as such

**Comment by Leont** [comment]
Let me rephrase it differently: what problem does including this in core solve? Is it just "one less dependency in your stack", or is there something more.

**Comment by Grinnz** [comment]
The benefit of including any module like this in core: implicit recommendation and appearance of providing a more complete toolkit. I don't really have an opinion on whether URI in particular is suitable - there are several more important tools we should consider first IMO.

**Comment by haarg** [comment]
I think URI has a quite poor interface, so I would not support including it in core.

**Comment by jkeenan** [comment]
In many legislative manuals of parliamentary procedure, there is a provision commonly referred to as "motion fails for lack of a second," *i.e.,* if no one other than a motion's original sponsor speaks up in favor of the motion, then the motion is dismissed.

Unfortunately, this feature request does not yet have a second and has already encountered some thumbs-down.

Nonetheless, thanks to @rwp0 for the original submission.  Closing ticket.

Thank you very much.
Jim Keenan
